iFixit/react-commerce
# Improve previews seed

closes #498 

## Changelog

- 🆕 Strapi plugin: iFixit addons. This new plugin is entirely written in typescript and replaces the old seed plugin. I gave the plugin a generic name because the intent is to use it as a place where we can put our custom made Strapi tools.
This initial implementation makes the plugin able to seed data using `strapi.ifixit.com` data.
- 🆙 Minor Strapi upgrade to v4.3.3 to fix some minor issues

👁️  **NOTE**: There are some changes to the Docker file that are necessary to build the typescript plugin. These changes have been tested to work with Govinor docker deployment, so I expect them to work in production, but it's worth to double check.

> ℹ️ Most of the stuff has been auto generated by Strapi using the `strapi generate` command. The interesting bits are in the folder `backend/src/plugins/addons/server/services/seed`

## QA

1. Visit the Vercel Preview
2. Verify that the preview is showing production-like data
3. Clone the project and install dependencies (make sure to follow install instructions from the readme)
4. Run `pnpm dev` and log into the local Strapi instance at http://localhost:1337/admin
    > ⚠️ If you already have the project cloned on your computer, chances are that the Strapi db is already initialized. If that's the case then reset it by deleting the file at `backend/.tmp/data.db`
6. Verify that the instance has a copy of production data
